WordPress/openverse-frontend
# Extract filter store from the search store and convert it to Pinia

## Fixes


Related to #1037 by @obulat

## Description


Search store was responsible for the state of filters, API query parameters, and conversions between them.

This PR extracts the state of filters to a separate store, and converts it to Pinia. The initial filter creation was extracted to `/constants/filters.js` in #1038.
The filter store actions are mostly used by the `searchStore`, and components use its `getters`.

## Testing Instructions

Run the app and see that search and filtering works as usual for all search type, both on the client side and on the server-rendered pages (do a hard page reload, or go directly to an address with a filter checked, such as `http://localhost:8443/search/audio/?q=sound&license=cc0&mature`, try going back using the browser back button.
Note: It won't be possible to test `mature` filter using the UI until #542 is fixed.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
